include/pipe.cc

```diff
@@ -1,26 +1,83 @@
-// Constructor with id and node pointers
+// Constructor with id, node pointers, diameter and status
 pipenetwork::Pipe::Pipe(
-    unsigned id, const std::array<std::shared_ptr<pipenetwork::Node>, 2>& nodes)
-    : id_{id}, nodes_{nodes} {
+    unsigned id, const std::array<std::shared_ptr<pipenetwork::Node>, 2>& nodes,
+    const double diameter, bool status)
```

Comment:
Can we not set `max_velocity` input argument to a default value, thereby avoiding 2 constructors?